src/robotFactory.js

```diff
@@ -1,10 +1,86 @@
 'use strict';
 
-class BaseRobot {}
+class BaseRobot {
+  constructor(name, weight, { x = 0, y = 0 }, chipVersion) {
+    this.name = name;
+    this.weight = weight;
 
-class FlyingRobot {}
+    this.coords = {
+      x: x,
+      y: y,
+    };
+    this.chipVersion = chipVersion;
+  }
 
-class DeliveryDrone {}
+  goForward(step = 1) {
+    this.coords.y += step;
+  }
+
+  goBack(step = 1) {
+    this.coords.y -= step;
+  }
+
+  goRight(step = 1) {
+    this.coords.x += step;
+  }
+
+  goLeft(step = 1) {
+    this.coords.x -= step;
+  }
+
+  getInfo() {
+    let info = `Robot: ${this.name}, Chip version: ${this.chipVersion},`;
+
+    info += ` Weight: ${this.weight}`;
+
+    return info;
```

Comment:
if you have any questions or know, that you do incorrect, you shouldn't create the PR, you should ask your question in the fe_chat
```suggestion
   return `Robot: ${this.name}, Chip version: ${this.chipVersion},`
     + ` Weight: ${this.weight}`;
```

Reply:
Thanks)